Earthfile

```diff
@@ -1,84 +1,85 @@
+build-environment:
+    FROM --pass-args +init
+    DO --pass-args +INSTALL_DEPS
+
+# configure :
+#   Installs deps and configures the project without building.
+configure:
+    FROM --pass-args +build-environment
+
+    # Various important paths for the build
+    ARG source_dir = $__source_dir
+    ARG build_dir  = $__build_dir
+
+    # Add the source tree into the container
+    DO +COPY_SOURCE --into=$source_dir
+
+    # Configure the project
     ARG --required tls
     ARG --required sasl
     RUN cmake -S "$source_dir" -B "$build_dir" -G "Ninja Multi-Config" \
         -D ENABLE_MAINTAINER_FLAGS=ON \
         -D ENABLE_SHM_COUNTERS=ON \
         -D ENABLE_SASL=$(echo $sasl | __str upper) \
-        -D ENABLE_SNAPPY=ON \
+        -D ENABLE_SNAPPY=AUTO \
         -D ENABLE_SRV=ON \
         -D ENABLE_ZLIB=BUNDLED \
         -D ENABLE_SSL=$(echo $tls | __str upper) \
-        -D ENABLE_COVERAGE=ON \
+        -D ENABLE_COVERAGE=OFF \
         -D ENABLE_DEBUG_ASSERTIONS=ON \
         -Werror
 
 # build :
-#   Build libmongoc and libbson using the specified environment.
+#   Install deps, configures, and builds libmongoc and libbson.
 #
-# The --env argument specifies the build environment among the `+env.xyz` environment
-# targets, using --purpose=build for the build environment. Refer to the target
-# list for a list of available build environments.
+#   Building this target requires certain arguments to be specified, the most important
+#   being `--from`, which specifies the base container image that is used for the build
 build:
-    # env is an argument
-    ARG --required env
-    FROM --pass-args +env.$env --purpose=build
-    # The configuration to be built
-    ARG config=RelWithDebInfo
-    # The prefix at which to install the built result
-    ARG install_prefix=/opt/mongo-c-driver
-    # Build configuration parameters. Will be case-normalized for CMake usage.
-    ARG --required sasl
-    ARG --required tls
-    LET source_dir=/opt/mongoc/source
-    LET build_dir=/opt/mongoc/build
-    DO +COPY_SOURCE --into=$source_dir
-    ENV CCACHE_HOME=/root/.cache/ccache
-    DO --pass-args +CONFIGURE --source_dir=$source_dir --build_dir=$build_dir
-    RUN --mount=type=cache,target=$CCACHE_HOME \
-        env CCACHE_BASE="$source_dir" \
-            cmake --build $build_dir --config $config
-    RUN cmake --install $build_dir --prefix="$install_prefix" --config $config
-    SAVE ARTIFACT /opt/mongoc/build/* /build-tree/
-    SAVE ARTIFACT /opt/mongo-c-driver/* /root/
+    ARG config          = "RelWithDebInfo"
+    LET install_prefix  = "/opt/mongo-c-driver"
+
+    # Do the configure step. Force the default value for $build_dir
+    FROM --pass-args +configure --build_dir=$__build_dir
+
+    # Run the add-ccache command here. This needs to run directly within the same
+    # target that makes use of it, to ensure that the CACHE line has an effect
+    # within that target
+    DO --pass-args +ADD_CCACHE
```

Comment:
Is there a point to adding ccache support within Earthly tasks? AFAIK the ccache cache directory will be isolated within each temporary container it is executed within, then immediately thrown away along with the temporary container. There is no opportunity for reuse unless an existing cache directory is copied/preserved across Earthly tasks, such as (I think) via `--mount type=cache`.